pkg/cloud/gcp/actuators/machine/reconciler_test.go

```diff
@@ -16,37 +19,154 @@ import (
 )
 
 func TestCreate(t *testing.T) {
-	_, mockComputeService := computeservice.NewComputeServiceMock()
-	machineScope := machineScope{
-		machine: &machinev1beta1.Machine{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      "",
-				Namespace: "",
-				Labels: map[string]string{
-					machinev1beta1.MachineClusterIDLabel: "CLUSTERID",
+	cases := []struct {
+		name                string
+		labels              map[string]string
+		providerSpec        *gcpv1beta1.GCPMachineProviderSpec
+		expectedCondition   *gcpv1beta1.GCPMachineProviderCondition
+		secret              *apiv1.Secret
+		mockInstancesInsert func(project string, zone string, instance *compute.Instance) (*compute.Operation, error)
+		expectedError       error
+	}{
+		{
+			name: "Successfully create machine",
+			expectedCondition: &gcpv1beta1.GCPMachineProviderCondition{
+				Type:    gcpv1beta1.MachineCreated,
+				Status:  corev1.ConditionTrue,
+				Reason:  machineCreationSucceedReason,
+				Message: machineCreationSucceedMessage,
+			},
+			expectedError: nil,
+		},
+		{
+			name: "Fail on invalid target pools",
+			providerSpec: &gcpv1beta1.GCPMachineProviderSpec{
+				TargetPools: []string{""},
+			},
+			expectedError: errors.New("failed validating machine provider spec: all target pools must have valid name"),
+		},
+		{
+			name: "Fail on invalid missing machine label",
+			labels: map[string]string{
+				machinev1beta1.MachineClusterIDLabel: "",
+			},
+			expectedError: errors.New("failed validating machine provider spec: machine is missing \"machine.openshift.io/cluster-api-cluster\" label"),
+		},
+		{
+			name: "Fail on invalid user data secret",
+			providerSpec: &gcpv1beta1.GCPMachineProviderSpec{
+				UserDataSecret: &corev1.LocalObjectReference{
+					Name: "notvalid",
 				},
 			},
+			secret: &apiv1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "notvalid",
+				},
+				Data: map[string][]byte{
+					"badKey": []byte(""),
+				},
+			},
+			expectedError: errors.New("error getting custom user data: secret /notvalid does not have \"userData\" field set. Thus, no user data applied when creating an instance"),
+		},
+		{
+			name:          "Fail on compute service error",
+			expectedError: errors.New("failed to create instance via compute service: fail"),
+			expectedCondition: &gcpv1beta1.GCPMachineProviderCondition{
+				Type:    gcpv1beta1.MachineCreated,
+				Status:  corev1.ConditionFalse,
+				Reason:  machineCreationFailedReason,
+				Message: "fail",
+			},
+			mockInstancesInsert: func(project string, zone string, instance *compute.Instance) (*compute.Operation, error) {
+				return nil, errors.New("fail")
+			},
+		},
+		{
+			name:          "Fail on google api error",
+			expectedError: errors.New("error launching instance: googleapi: Error 400: error"),
```

Comment:
We are returning a specific type of error in the function tested.  We should use that type.